### PR TITLE
✨ Efs Guardian can create directories (OPS ⚠️)

### PR DESCRIPTION
**Pull request by matusdrobuliak66**
## What do these changes do?
- ✨ Efs Guardian can create directories
  -  Will be used by director-v2 before starting sidecar with mounted volume pointed to the project specific directory in the distributed file system

Next steps (Efs-Guardian responsibilities):
- Change user permissions on the folder if he goes over some threshold
- Remove data if it is not used

## Related issue/s
- relates https://github.com/ITISFoundation/osparc-issues/issues/1313
- relates https://github.com/ITISFoundation/osparc-issues/issues/1333
- relates https://github.com/ITISFoundation/osparc-simcore/issues/5872
- prerequisite for https://github.com/ITISFoundation/osparc-simcore/pull/5873

## How to test
`test_rpc_create_project_specific_data_dir`

## Dev-ops checklist
- ⚠️ new env var: `EFS_DNS_NAME` --> This variable was already added in the osparc-master-zmt.click :heavy_check_mark: 
- ⚠️ new optional `EFS_MOUNTED_PATH`
- [x] No ENV changes or I properly updated ENV ([read the instruction](https://git.speag.com/oSparc/osparc-ops-deployment-configuration/-/blob/configs/README.md?ref_type=heads#how-to-update-env-variables))

### CODE REVIEW

File: services/efs-guardian/src/simcore_service_efs_guardian/core/settings.py [hunk 1]
```diff
@@ -69,6 +71,7 @@ class ApplicationSettings(BaseCustomSettings, MixinLoggingSettings):
     EFS_GUARDIAN_AWS_EFS_SETTINGS: AwsEfsSettings | None = Field(
         auto_default_from_env=True
     )
+    EFS_GUARDIAN_RABBITMQ: RabbitSettings | None = Field(auto_default_from_env=True)
```
Comment (sanderegg) [inline services/efs-guardian/src/simcore_service_efs_guardian/core/settings.py:74]: question: does it make sense to make this optional? e.g. can it work without RabbitMQ? actually same question for the EFS settings above? they should not be optional no?

File: services/efs-guardian/src/simcore_service_efs_guardian/core/settings.py [hunk 2]
```diff
@@ -22,7 +23,8 @@ class AwsEfsSettings(BaseCustomSettings):
         description="AWS Elastic File System DNS name",
         example="fs-xxx.efs.us-east-1.amazonaws.com",
     )
-    EFS_BASE_DIRECTORY: str = Field(default="project-specific-data")
+    EFS_PROJECT_SPECIFIC_DATA_DIRECTORY: str = Field(default="project-specific-data")
+    EFS_MOUNTED_PATH: str = Field(default="/data/efs")
```
Comment (sanderegg) [inline services/efs-guardian/src/simcore_service_efs_guardian/core/settings.py:27]:
```suggestion
    EFS_MOUNTED_PATH: Path = Field(default="/data/efs")
```

File: services/efs-guardian/src/simcore_service_efs_guardian/services/efs_manager.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+from fastapi import FastAPI
+from models_library.projects import ProjectID
+from models_library.projects_nodes_io import NodeID
+
+
+@dataclass(frozen=True)
+class EfsManager:
+    app: FastAPI
+
+    _efs_mounted_path: str
```
Comment (sanderegg) [inline services/efs-guardian/src/simcore_service_efs_guardian/services/efs_manager.py:13]:
```suggestion
    _efs_mounted_path: Path
```

File: services/efs-guardian/src/simcore_service_efs_guardian/services/efs_manager.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    _project_specific_data_base_directory: str
+
+    @classmethod
+    async def create(
+        cls,
+        app: FastAPI,
+        efs_mounted_path: str,
```
Comment (sanderegg) [inline services/efs-guardian/src/simcore_service_efs_guardian/services/efs_manager.py:20]:
```suggestion
        efs_mounted_path: Path,
```

File: services/efs-guardian/src/simcore_service_efs_guardian/services/efs_manager.py [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+        project_specific_data_base_directory: str,
```
Comment (sanderegg) [inline services/efs-guardian/src/simcore_service_efs_guardian/services/efs_manager.py:21]:
```suggestion
        project_specific_data_base_directory: Path,
```
is it a directory or just a name?

File: services/efs-guardian/src/simcore_service_efs_guardian/services/efs_manager.py [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+    ):
+        return cls(app, efs_mounted_path, project_specific_data_base_directory)
+
+    async def initialize_directories(self):
+        _dir_path = (
+            Path(self._efs_mounted_path) / self._project_specific_data_base_directory
```
Comment (sanderegg) [inline services/efs-guardian/src/simcore_service_efs_guardian/services/efs_manager.py:27]: I believe that with the modifications above, you can then remove this cast

File: services/efs-guardian/src/simcore_service_efs_guardian/services/efs_manager.py [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+        )
+        Path.mkdir(_dir_path, parents=True, exist_ok=True)
+
+    async def create_project_specific_data_dir(
+        self, project_id: ProjectID, node_id: NodeID
+    ) -> Path:
+        _dir_path = (
+            Path(self._efs_mounted_path)
```
Comment (sanderegg) [inline services/efs-guardian/src/simcore_service_efs_guardian/services/efs_manager.py:35]: same here

File: services/efs-guardian/tests/unit/test_api_health.py [hunk 8]
```diff
@@ -3,10 +3,14 @@
...
+from settings_library.rabbit import RabbitSettings
 from starlette import status
 
+pytest_simcore_core_services_selection = ["rabbit"]
+pytest_simcore_ops_services_selection = []
 
-async def test_healthcheck(client: httpx.AsyncClient):
+
+async def test_healthcheck(rabbit_service: RabbitSettings, client: httpx.AsyncClient):
     response = await client.get("/")
```
Comment (sanderegg) [inline services/efs-guardian/tests/unit/test_api_health.py:14]: for the healthcheck, I think if you look in autoscaling/clusters-keeper I make the healthcheck go bad when connection to dependencies is failing. and I think you need to add the rabbitmq issue as well there no?

**Review by sanderegg — APPROVED** [review]
Looking good! please consider my comments. thanks!

**Review by GitHK — APPROVED** [review]
looking good

File: services/efs-guardian/src/simcore_service_efs_guardian/services/efs_manager.py (hunk 5 above)
Comment (matusdrobuliak66) [reply services/efs-guardian/src/simcore_service_efs_guardian/services/efs_manager.py:21]: Its a directory name in the root directory of the mounted file system

File: services/efs-guardian/src/simcore_service_efs_guardian/services/efs_manager_setup.py [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+
+    async def on_shutdown() -> None:
+        if app.state.efs_manager:
+            ...
+
+    app.add_event_handler("startup", on_startup)
+    app.add_event_handler("shutdown", on_shutdown)
+
+
+def get_efs_manager(app: FastAPI) -> EfsManager:
```
Comment (pcrespov) [inline services/efs-guardian/src/simcore_service_efs_guardian/services/efs_manager_setup.py:49]: Checkout `SingletonInAppStateMixin` to get/set some object in the `app.state`

**Review by pcrespov — APPROVED** [review]

File: packages/service-library/src/servicelib/rabbitmq/rpc_interfaces/efs_guardian/efs_manager.py [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+@log_decorator(_logger, level=logging.DEBUG)
+async def create_project_specific_data_dir(
+    rabbitmq_rpc_client: RabbitMQRPCClient,
+    *,
+    project_id: ProjectID,
+    node_id: NodeID,
+) -> Path:
+    output: Path = await rabbitmq_rpc_client.request(
+        EFS_GUARDIAN_RPC_NAMESPACE,
+        parse_obj_as(RPCMethodName, "create_project_specific_data_dir"),
```
Comment (mrnicegyu11) [inline packages/service-library/src/servicelib/rabbitmq/rpc_interfaces/efs_guardian/efs_manager.py:29]: minor: the word `specific` is redundant imho

**Review by mrnicegyu11 — APPROVED** [review]
lgtm

File: services/efs-guardian/tests/unit/test_api_health.py (hunk 8 above)
Comment (matusdrobuliak66) [reply services/efs-guardian/tests/unit/test_api_health.py:32]: I will modify the health check mechanism in different PR (I need to think/discuss a bit about it)

File: packages/service-library/src/servicelib/rabbitmq/rpc_interfaces/efs_guardian/efs_manager.py (hunk 10 above)
Comment (matusdrobuliak66) [reply packages/service-library/src/servicelib/rabbitmq/rpc_interfaces/efs_guardian/efs_manager.py:29]: The idea here was, that there might be in the future for example `project-shared-data`

**Pull request merged**